extensions/binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java

```diff
@@ -127,21 +142,66 @@ public void handleCommand(ChannelUID channelUID, Command command) {
         }
     }
 
+    @Override
+    public Collection<ConfigStatusMessage> getConfigStatus() {
+        Collection<ConfigStatusMessage> configStatus = new ArrayList<>();
+
+        try {
+            String weatherData = getWeatherData();
+            String result = StringUtils.substringBetween(weatherData, "<item><title>", "</title>");
+            if ("City not found".equals(result)) {
+                configStatus.add(ConfigStatusMessage.Builder.error(LOCATION_PARAM, LOCATION_NOT_FOUND)
+                        .withArguments(location).build());
+            }
+        } catch (IOException e) {
+            logger.debug("Communication error occurred while getting Yahoo weather information.", e);
+        }
+
+        return configStatus;
+    }
+
+    @Override
+    public boolean supportsEntity(String entityId) {
+        return getThing().getUID().getAsString().equals(entityId);
+    }
+
+    @Override
+    public void setConfigStatusCallback(ConfigStatusCallback configStatusCallback) {
+        this.configStatusCallback = configStatusCallback;
+    }
+
+    @Override
+    public void handleConfigurationUpdate(Map<String, Object> configurationParameters)
```

Comment:
I understand your points... But let us imagine a thing handler is a config status provider and (in future) a FirmwareUpdateHandler as proposed by issue #1153 .... Following your suggestion then we would have to add an additional "FirmwareUpdateThingHandler" implementing the base operations for the FirmwareUpdateHandler.... And if a ThingHandler is both a ConfigStatusProvivder and a FirmwareUpdateHandler what is your suggestion to solve this problem? Shall we offer a base thing handler extension implementing both the ConfigStatusProvider and in future the FirmwareUpdateHandler?